packages/sanity/src/core/store/_legacy/document/document-pair/memoizedPair.ts

```diff
@@ -15,5 +15,11 @@ export const memoizedPair: (
       subscriber.next(checkoutPair(client, idPair))
     }).pipe(publishReplay(1), refCount())
   },
-  (_client, idPair) => idPair.publishedId
+  (client: SanityClient, idPair: IdPair, typeName?: string) => {
+    const config = client.config()
+
+    return `${config.dataset ?? ''}-${config.projectId ?? ''}-${idPair.publishedId}-${
+      typeName ?? ''
```

Comment:
I might be missing something, but since IDs are always unique and two documents of different types can't share the same ID, I'm not sure if we need to have `typeName` as part of the key here?

Reply:
A document can be deleted and recreated with a different type. Would this create problems? I feel like its safer to include the typename.

Reply:
Ah, good point, and that's something that happens when people run migration scripts renaming types, so not that rare either. Talked with @binoy14 yesterday and it seems like it's straightforward to make `typeName` required, so we will use it as part of the cache key from now on. Thanks!